Haskell/Haskell.sublime-syntax

```diff
@@ -157,19 +157,14 @@ contexts:
         4: constant.character.escape.hexadecimal.haskell
         5: constant.character.escape.control.haskell
         6: punctuation.definition.string.end.haskell
-    - match: '^\s*([a-z_][a-zA-Z0-9_'']*|\([|!%$+\-.,=</>]+\))\s*(::|∷)'
+    - match: '^(\s*)([a-z_][a-zA-Z0-9_'']*|\([|!%$+\-.,=</>]+\))\s*(::|∷)'
       captures:
-        1: entity.name.function.haskell
-        2: keyword.other.double-colon.haskell
+        2: entity.name.function.haskell
+        3: keyword.other.double-colon.haskell
       push:
         - meta_scope: meta.function.type-declaration.haskell
-        - match: |-
-            (?x)
-                ^(data|newtype|type|class|deriving)\s  # When a top level declaration starts
-              | ^[^=]*(=)[\sa-zA-Z0-9_\(]                # A function declaration
-          captures:
-            1: keyword.other.haskell
-            2: keyword.operator.haskell
+        - include: comments
+        - match: ^(?!\1\s)
```

Comment:
You may want to ignore line comments here, too, unless the comments include above also starts from the first character of the line and would thus take precedence. (I'm in mobile.)

Reply:
The `comments` context includes single-line comments and the `block_comment` context. None of their patterns are anchored to the beginning of the line. Won't the comments always take precedence because of the position of the definition?

From [Sublime's documentation](https://www.sublimetext.com/docs/3/syntax.html#contexts): "When multiple patterns match at the same position, the first defined pattern will be selected."

Reply:
The pattern I annodated here matches on the first character of a line (due to the `^` anchor) while the comment pattern would only match on the column the comment begins. You should be able to test this by having a comment with less leading whitespace than the original indent where this context should not be popped.

Reply:
Apologies for the confusion. Is there a way to ignore comments without producing overlapping `meta_scope`'s?

Reply:
What kind of meta scope are you looking to remove? The meta for a function definition? I don't think that matters, or any similar scope. I mean, we are still within that conceptually and it's just a comment interweaved.

Reply:
I was trying to avoid having a comment that has both `meta.function.type-declaration.haskell` and `comment.line.double-dash.haskell`. This would occur if I replaced the popping pattern with `^(?!\s*--|\1\s)`. I guess this is a nonproblem. It just thought it made more sense to think of a comment that splits a type signature as an interruption of the signature, as opposed to being contained within it.

Reply:
In all other syntaxes we consider comments to be extaneous and they may be appear anywhere they are legal. There is no splitting going on, it's just some dust lingering around.